### ISSUE TITLE
Proposal: enhance JSDoc deprecate parameter description

### ISSUE THREAD

**Issue by xDivisionByZerox** [faker-js/faker#976]
### Clear and concise description of the problem

As a developer using faker I want to see some intel about a deprecated method in intellisense. This could help me to figure out what I should use instead without waiting for a log that tells me.

### Suggested solution

Add meaningful documentation to JSDocs `deprecated` parameter.
 
Like:
```
/**
  * @deprecated
  * Use faker.name.firstName instead.
  * /
faker.name.firstname()
```
This would result in a more excellent information box resulting and in a better developer experience by not only telling me that a method is deprecated but telling me what I should do instead.
Like:
![image](https://user-images.githubusercontent.com/77127505/169600210-89bb69c2-224f-4659-815e-622dd57bf977.png)

### Alternative

Don't do anything

### Additional context

[rxjs is doing the same for their operators](https://github.com/ReactiveX/rxjs/blob/d25b3528a4168c9ba76f81ec84ee20f6c241903b/src/internal/operators/combineLatest.ts) (just ONE reference)

**Comment by ST-DDT** [comment]
Should we include that also in our api docs page?

**Comment by xDivisionByZerox** [comment]
> Should we include that also in our api docs page?

I think that would be a great idea. I don't use the API docs TBH so I didn't realize they could be in need of this feature as well.

**Comment by import-brain** [comment]
> Should we include that also in our api docs page?

Yeah, I also think this would be a great idea.

**Comment by xDivisionByZerox** [comment]
Currently, we provide this information via the `@see` parameter. Would we move this into the `@deprecated` parameter description and no longer use it (`@see`) for deprecations?

**Comment by import-brain** [comment]
> Currently, we provide this information via the `@see` parameter. Would we move this into the `@deprecated` parameter description and no longer use it (`@see`) for deprecations?

Yeah, I think it would be more clear if we moved the information to the `@deprecated` parameter and wrote something like "Use this function instead".

**Comment by ST-DDT** [comment]
If it generates a clickable link for the docs, sure go for it.

**Comment by xDivisionByZerox** [comment]
Do you mean in the docs? Because currently there is no link in the JSDoc in my IDE :/

**Comment by ST-DDT** [comment]
I'm referring to the api docs website:
Example: https://fakerjs.dev/api/address.html#cityPrefix

**Comment by Shinigami92** [comment]
Here an image, as the link posted above by @ST-DDT will get stale some day

<img width="400" alt="image" src="https://user-images.githubusercontent.com/7195563/173526616-cd3d8edd-e78e-413d-8e0d-09d04e6505b9.png">

**Comment by Shinigami92** [comment]
Repeating my message here from: https://github.com/faker-js/faker/pull/1067#discussion_r896490442

> I would say @see => links to the replacement, @deprecation message => message = a human readable text about what's going on 🤔

**Comment by xDivisionByZerox** [comment]
This has been resolved by #1067.
